src/m365/spo/commands/applicationcustomizer/applicationcustomizer-remove.ts

```diff
@@ -0,0 +1,212 @@
+    const { webUrl } = options;
+    if (webUrl) {
+      const isValidSharePointUrl = validation.isValidSharePointUrl(webUrl);
+      if (isValidSharePointUrl !== true) {
+        return isValidSharePointUrl;
+      }
+    }
+    return true;
+  }
+
+  private async validateParams({ options }: CommandArgs): Promise<boolean | string> {
+    const { id, clientSideComponentId, title } = options;
+    if (id || clientSideComponentId || title) {
+      return true;
+    }
+    else {
+      return `At least one of the parameters id, clientSideComponentId, or title must have a value`;
+    }
+  }
+
+  private async validateGUIds({ options }: CommandArgs): Promise<boolean | string> {
+    const { clientSideComponentId, id } = options;
+    if (clientSideComponentId && !validation.isValidGuid(clientSideComponentId)) {
+      return `${clientSideComponentId} is not a valid GUID`;
+    }
+
+    if (id && !validation.isValidGuid(id)) {
+      return `${id} is not a valid GUID`;
+    }
+    return true;
+  }
+
+  private async validateScope(args: CommandArgs): Promise<boolean | string> {
+    const { options } = args;
+    const { scope } = options;
+    if (scope && SpoApplicationCustomizerRemoveCommand.scopes.indexOf(scope) < 0) {
+      return `${scope} is not a valid value for scope. Valid values are ${SpoApplicationCustomizerRemoveCommand.scopes.join(', ')}`;
+    }
+    return true;
+  }
+
+  private async getAppCustomizers(args: CommandArgs): Promise<CustomAction[]> {
+    const { options }: CommandArgs = args;
+    const { id, webUrl, title, clientSideComponentId, scope }: Options = options;
+    const appCustomizers = await spo.getCustomActions(webUrl, scope, `Location eq 'ClientSideExtension.ApplicationCustomizer'`);
+    const filteredAppCustomizers: CustomAction[] = appCustomizers.filter(appCustomizer =>
+      (id && appCustomizer.Id.includes(id)) ||
+      (!id && appCustomizer.Title.includes(`${title}`)) ||
+      (!id && appCustomizer.ClientSideComponentId.includes(`${clientSideComponentId}`))
+    );
+    return filteredAppCustomizers;
+  }
+
+  private getAppCustomizerToRemove(customActions: CustomAction[]): CustomAction {
+    const customActionsCount = customActions.length;
+    if (customActionsCount === 0) {
+      throw `No application customizer found`;
+    }
+    if (customActionsCount > 1) {
+      const ids = customActions.map(a => a.Id).join(', ');
+      throw `Multiple application customizer found. Please disambiguate using IDs: ${ids}`;
+    }
+    return customActions[0];
+  }
+
+  private removeAppCustomizer(scope: string, webUrl: string, id: string): Promise<void> {
+    const requestOptions: any = {
+      url: `${webUrl}/_api/${scope}/UserCustomActions('${id}')`,
+      headers: {
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+    return request.delete(requestOptions);
+  }
+
+  private async confirmPrompt({ options }: CommandArgs): Promise<boolean> {
```

Comment:
I find this function a bit verbose. If you compare it with `commandset-remove.ts` you'll see you're using a lot of lines that are not really necessary. Could you try to refactor?